src/types.ts

```diff
@@ -113,6 +117,22 @@ export interface PredefinedType extends BaseValueType {
   name: string;
 }
 
+export interface LiteralType extends BaseValueType {
+  kind: ValueTypeKind.literalType;
```

Comment:
I don't find it's necessary for `LiteralType` to have its `kind`. It looks like `LiteralType` is more like an internal convent type definition like `EnumField`.

Reply:
I added kind for BaseValueType, but it really isn't used. Let me remove it

Reply:
It doesn't have to extend `BaseValueType`.

Reply:
yes, I will also remove BaseValueType